pkg/security/secl/compiler/eval/variables.go

```diff
@@ -485,13 +484,9 @@ func (v *ScopedVariables[T]) GetVariable(name string, value interface{}) (Variab
 		key := v.scoper(ctx)
 		vars := v.vars[key]
 		if vars == nil {
-			vars = &Variables{}
-			v.vars[key] = vars
-			if v.onNewVariables != nil {
-				v.onNewVariables(key)
-			}
+			v.vars[key] = &Variables{}
 		}
-		vars.Set(name, value)
+		v.vars[key].Set(name, value)
```

Comment:
I think this fixes a bug, where `vars` before would point to nil and not the new inserted variables